libs/gl-client-py/glclient/lsps.py

```diff
@@ -91,29 +91,32 @@ class NoParams:
 
 
 class LspClient:
-    def __init__(self, native: native.LspClient, peer_id: t.Union[bytes, str]):
+    def __init__(self, native: native.LspClient):
         self._native = native
-        self._peer_id: bytes = parse_and_validate_peer_id(peer_id)
 
     def _rpc_call(
         self,
-        peer_id: bytes,
+        peer_id: str,
         method_name: str,
         param_json: bytes,
         json_rpc_id: t.Optional[str] = None,
     ) -> bytes:
         logger.debug("Request lsp to peer %s and method %s", peer_id, method_name)
+        peer_id_bytes = bytes.fromhex(peer_id)
```

Comment:
Do you think we should still use `parse_and_validate_peer_id`? If so, it might be good to call it here

Reply:
I agree that removing `parse_and_validate_peer_id` is appropriate.

In my initial design I used a `bytes`-object to represent a node-id. 
This was annoying because the JSON-RPC uses a hex-encoded string.
The goal of  `parse_and_validate_peer_id` was to ensure we could accept both versions.

This method is obsolete now and I'll remove it